File: web-api/src/main/java/cn/labzen/web/api/annotation/Abandoned.java
package cn.labzen.web.api.annotation;

import java.lang.annotation.*;

/**
 * 标识 Controller 接口方法已弃用的注解。
 * <p>
 * 使用此注解标记的接口方法将不再作为增删改查的 Restful API 入口。
 * 生成的 Controller 实现类中，该方法将被排除在外。
 */
@Documented
@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.METHOD)
public @interface Abandoned {
}

File: src/main/kotlin/cn/labzen/web/annotation/Abandoned.kt
package cn.labzen.web.annotation

/**
 * 注解在 Controller 接口方法上，标识该方法弃用，将不再作为增删改查的 Restful API 入口
 */
@MustBeDocumented
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.SOURCE)
annotation class Abandoned